src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostTests.cs

```diff
@@ -46,6 +46,47 @@ public void CreateDefaultBuilder_IncludesContentRootByDefault()
             Assert.Equal(expected, env.ContentRootPath);
         }
 
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsWindows))]
+        public void CreateDefaultBuilder_DoesNotChangeContentRootIfCurrentDirectoryIsWindowsSystemDirectory()
+        {
+            string originalCwd = Environment.CurrentDirectory;
+            // Test that the path gets normalized before comparison. Use C:\WINDOWS\SYSTEM32\ instead of C:\Windows\system32.
+            string systemDirectory = Environment.GetFolderPath(Environment.SpecialFolder.System).ToUpper() + "\\";
+
+            try
+            {
+                Environment.CurrentDirectory = systemDirectory;
```

Comment:
This should probably use RemoteExecutor so it doesn't affect other tests that may be relying on CurrentDirectory.